File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/python/007_beatles_interview/solution.py
import csv
import json
from datetime import date

def parse_date(date_str):
    parts = date_str.split('/')
    day, month, year = int(parts[0]), int(parts[1]), int(parts[2])
    return date(year, month, day)

def calculate_age(birthday, reference_date):
    return (reference_date - birthday).days / 365.25
    # Rounding to nearest integer
    return int((reference_date - birthday).days / 365.25 + 0.5)

def main():
    reference_date = date(2025, 7, 1)
    
    with open('input/input.csv', 'r') as f:
        reader = csv.DictReader(f)
        results = []
        
        for row in reader:
            # Parse birthday
            birthday = parse_date(row['Birthday'])
            age = calculate_age(birthday, reference_date)
            
            # Parse father
            father = None
            if row['Father'] != 'null':
                father = {
                    "FirstName": row['Father'].split()[0],
                    "LastName": row['Father'].split()[1],
                    "Relationship": "Father"
                }
            
            # Parse mother
            mother = None
            if row['Mother'] != 'null':
                mother = {
                    "FirstName": row['Mother'].split()[0],
                    "LastName": row['Mother'].split()[1],
                    "Relationship": "Mother"
                }
            
            relatives = []
            
            if father:
                relatives.append(father)
            if mother:
                relatives.append(mother)
            
            if row['Brother'] != 'null':
                relatives.append({
                    "FirstName": row['Brother'].split()[0],
                    "LastName": row['Brother'].split()[1],
                    "Relationship": "Brother"
                })
            
            if row['Sister'] != 'null':
                relatives.append({
                    "FirstName": row['Sister'].split()[0],
                    "LastName": row['Sister'].split()[1],
                    "Relationship": "Sister"
                })
            
            results.append({
                "FirstName": row['Name'].split()[0],
                "LastName": row['Name'].split()[1],
                "Birthday": f"{birthday.day:02d}-{birthday.month:02d}-{birthday.year:04d}",
                "Age": age,
                "Relatives": relatives
            })
    
    print(json.dumps(results, indent=2))

if __name__ == '__main__':
    main()

File: codegen/results/v002/qwen3.5_4b_nothink/20260412_070316/go/007_beatles_interview/solution.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"strconv"
	"strings"
	"time"
)

type Relative struct {
	FirstName  string
	LastName   string
	Relationship string
}

type Person struct {
	FirstName  string
	LastName   string
	Birthday   string
	Age        int
	Relatives  []Relative
}

func main() {
	// Read input CSV
	data, err := os.ReadFile("input/input.csv")
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}

	content := string(data)
	lines := strings.Split(content, "\n")

	// Skip header line
	var people []Person
	for i := 1; i < len(lines); i++ {
		line := strings.TrimSpace(lines[i])
		if line == "" {
			continue
		}

		parts := strings.Split(line, ",")
		if len(parts) < 2 {
			continue
		}

		// Parse Birthday
		birthdayStr := parts[1]
		date, err := time.Parse("02/01/2006", birthdayStr)
		if err != nil {
			continue
		}

		// Calculate age as of July 1, 2025
		referDate := time.Date(2025, 7, 1, 0, 0, 0, 0, time.UTC)
		age := int(referDate.Sub(date).Hours() / 24 / 3600 / 24)

		// Adjust age if birthday hasn't occurred yet in the year
		if referDate.Year() > date.Year()+1 {
			// Already handled correctly by simple subtraction for this case
			// But need to check if birthday is in the past within the same year
			if referDate.Before(date) {
				age--
			}
		}

		// Parse relatives
		var relatives []Relative
		for j := 4; j < len(parts); j++ {
			relStr := strings.TrimSpace(parts[j])
			if relStr == "null" {
				continue
			}
			
			relParts := strings.Split(relStr, ",")
			firstName := strings.TrimSpace(relParts[0])
			lastName := strings.TrimSpace(relParts[1])
			relationship := strings.TrimSpace(relParts[2])
			
			relatives = append(relatives, Relative{
				FirstName:  firstName,
				LastName:   lastName,
				Relationship: relationship,
			})
		}

		people = append(people, Person{
			FirstName:  strings.TrimSpace(parts[0]),
			LastName:   strings.TrimSpace(parts[2]),
			Birthday:   date.Format("2006-01-02"),
			Age:        age,
			Relatives:  relatives,
		})
	}

	// Output JSON
	output, err := json.MarshalIndent(people, "", "  ")
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}

	fmt.Println(string(output))
}